### PR TITLE
Objective C snippets generation

### LINKED ISSUE #33: Objective C Code Generation

### PR DESCRIPTION
**Pull request by andrueastman**
This pull request involves the following changes and closes #33 
- Objective C snippets generation with API for all Http methods with object construction
- Functions UppercaseFirstLetter and LowerCaseFirstLetter moved to CommonGenerator from CSharpGenerator to avoid code repetition
- Updated Readme to reflect added language
- Added unit tests to validate snippet generation
- Cleaned up csproj file.

Snippets can be previewed at this [docs preview](https://review.docs.microsoft.com/en-us/graph/api/user-list?view=graph-rest-1.0&branch=andrueastman%2Fsnippets-preview&tabs=objective-c)

### CODE REVIEW

File: CodeSnippetsReflection/LanguageGenerators/ObjectiveCGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+using System.Text;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+[assembly: InternalsVisibleTo("CodeSnippetsReflection.Test")]
+namespace CodeSnippetsReflection.LanguageGenerators
+{
+    public static class ObjectiveCGenerator
```
Comment (MIchaelMainer) [inline CodeSnippetsReflection/LanguageGenerators/ObjectiveCGenerator.cs]: Not review, just a question. Why are we creating an ObjC generator before Swift? My understanding is that we created our client lib in ObjC as it is easier to use in both ObjC and Swift projects. With that said, all new develop is highly encouraged to happen in Swift.

**Comment by andrueastman** [comment]
Sorry about the confusion but this PR is based off a branch and not a fork. (Though the branch name is prefixed by my username).

Easiest way to send a request manually to generate a snippet would be to use postman and send a request to the endpoint with the Graph HTTP request message in the request body. You can easily copy valid examples from the request sections in the example files [here](https://github.com/microsoftgraph/microsoft-graph-explorer-api/blob/dev/javascript-examples.md) and [here](https://github.com/microsoftgraph/microsoft-graph-explorer-api/blob/dev/c-sharp-examples.md). 
Hope this helps and will be happy to help with any more of this.

**Comment by MIchaelMainer** [comment]
Sorry for not looking more closely at that the PR source. 

What about the csproj pacakgereference udpate?

**Comment by andrueastman** [comment]
The csproj package reference is resolved. The graph package is not really used and was removed.

**Pull request merged**